src/classes/window/presenter.vala

```diff
@@ -571,11 +571,11 @@ namespace pdfpc.Window {
          * for display, as it is a Image widget after all.
          */
         public void set_cache_observer(CacheStatus observer) {
-            var current_prerendering_view = this.current_view as View.Prerendering;
+            var current_prerendering_view = this.current_view as View.Base;
             if (current_prerendering_view != null) {
                 observer.monitor_view(current_prerendering_view);
             }
-            var next_prerendering_view = this.next_view as View.Prerendering;
+            var next_prerendering_view = this.next_view as View.Base;
```

Comment:
`this.next_view` already is a `View.Base`, neither the casting nor the `null`-checking should be required.